refactor(storage): remove v1 downgrades from integration test read paths

SoumyaRaikwar opened the pull request:
## Which problem is this PR solving?
- Resolves #7050 (Part 1/2)
- Removes v1 downgrades from integration test read paths to prepare for full OTLP migration (in progress)
- Fixes deprecated Elasticsearch client usage
- Fixes lint errors (depguard, context usage)

## Description of the changes
- Refactored `integration` package tests to use `t.Context()` instead of `context.Background()` for better test timeout handling.
- Replaced deprecated `IndexTemplateExists` and `IndexDeleteTemplate` in Elasticsearch tests with direct API calls.
- Implemented `CreateDependencyWriter` in Cassandra v2 factory (shim to v1) to support dependency storage tests.
- Fixed `goleak` issues in tests.
- Removed unused `nolint` directives.
- Minor fixes in `trace_compare.go` and `v1adapter/translator.go`.

## How was this change tested?
- All integration tests passing.
- `make lint-go` passing.

## Checklist
- [x] I have read https://github.com/jaegertracing/jaeger/blob/master/CONTRIBUTING_GUIDELINES.md
- [x] I have signed all commits
- [x] I have added unit tests for the new functionality (N/A - refactoring existing tests)
- [x] I have run lint and test steps successfully
  - for `jaeger`: `make lint test`

SoumyaRaikwar commented:
@yurishkuro  could you please review my pr

SoumyaRaikwar commented on internal/storage/integration/trace_compare_otlp.go:
```diff
@@ -0,0 +1,75 @@
+}
+
+// CollectOTLPTraces collects all traces from an iterator into a slice
+func CollectOTLPTraces(iterTraces iter.Seq2[[]ptrace.Traces, error]) ([]ptrace.Traces, error) {
```
Thanks for the pointers
I’ve updated the storage integration tests to reuse the shared helpers:

- `jiter.FlattenWithErrors` for collecting from `iter.Seq2`
- `GetFirstTrace` / `OTLPTracesToV1Slice` for converting to v1 right before comparison

Tests no longer define their own collector, and variable names are back to neutral (`traces` / `actual`).

SoumyaRaikwar commented on internal/storage/integration/integration.go:
```diff
@@ -324,29 +333,35 @@ func (s *StorageIntegration) testGetTrace(t *testing.T) {
 		}
-		actual = traces[0]
-		return len(actual.Spans) == len(expected.Spans)
+		actualOTLP = otlpTraces[0]
```
Renamed back to `traces` / `actual` and now only convert to v1 right before comparison. No new `otlp*` names in the tests anymore.

SoumyaRaikwar commented:
@yurishkuro could you please review

yurishkuro commented on internal/storage/integration/integration.go:
```diff
@@ -193,12 +195,13 @@ func (s *StorageIntegration) testGetServices(t *testing.T) {
 					StartTimeMax: time.Now(),
 				})
-				traces, err := v1adapter.V1TracesFromSeq2(iterTraces)
+				traces, err := jiter.FlattenWithErrors(iterTraces)
```
what is the point of this change? How is this better than calling `v1adapter.V1TracesFromSeq2`?

SoumyaRaikwar commented:
Regarding Issue #7050

Upgrade Read Path: This PR (#7761) achieves this by consuming OTLP iterators directly from storage.
Upgrade Write Path: PR #7812 (@Manik2708) appears to address this by refactoring tests to write ptrace.Traces directly to the TraceWriter.
Remove Downgrades: Currently, we still downgrade to V1 for assertions.
Once #7812 done, it will functionally unblock the "Remove unnecessary downgrades" goal by allowing us to perform assertions directly on ptrace.Traces (preserving high-fidelity OTLP data like Scope attributes that V1 loses).

How would you like me to coordinate? I can pause till #7812 get merged , which seems to remain the critical piece for completing the full V2 transition described in the issue.

SoumyaRaikwar committed:
refactor: clean up integration tests and remove v1 adapter usage

Fix phantom redeclaration error by cleaning up generated files.

Implement CreateDependencyWriter in Cassandra V2 factory and refactor integration tests to use ptrace.Traces directly.

Signed-off-by: SoumyaRaikwar <somuraik@gmail.com>

SoumyaRaikwar committed:
Fix deprecated ES usage and lint errors

Signed-off-by: SoumyaRaikwar <somuraik@gmail.com>

yurishkuro commented:
do we still need this?

Manik2708 commented:
> do we still need this?

I think we can close this, as the issue reffered is closed
